File: VB/A2404/CodeFile1.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpo

Namespace DXSample
	Public Class User
		Inherits XPObject
		Private name_Renamed As String
		Public Sub New()
			MyBase.New()
		End Sub
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Public Property Name() As String
			Get
				Return name_Renamed
			End Get
			Set(ByVal value As String)
				name_Renamed = value
			End Set
		End Property
		<Association("User-Role", GetType(Role))> _
		Public ReadOnly Property Roles() As XPCollection
			Get
				Return GetCollection("Roles")
			End Get
		End Property
	End Class

	Public Class Role
		Inherits XPObject
		Public Sub New()
			MyBase.New()
		End Sub
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub

		Private roleName_Renamed As String
		Public Property RoleName() As String
			Get
				Return roleName_Renamed
			End Get
			Set(ByVal value As String)
				roleName_Renamed = value
			End Set
		End Property

		<Association("User-Role", GetType(User))> _
		Public ReadOnly Property Users() As XPCollection
			Get
				Return GetCollection("Users")
			End Get
		End Property
	End Class
End Namespace

File: CS/A2404/CodeFile1.cs
using DevExpress.Xpo;

namespace DXSample {
    public class User : XPObject {
        string name;
        public User() : base() { }
        public User(Session session) : base(session) { }
        public string Name { get { return name; } set { name = value; } }
        [Association("User-Role", typeof(Role))]
        public XPCollection Roles { get { return GetCollection("Roles"); } }
    }

    public class Role : XPObject {
        public Role() : base() { }
        public Role(Session session) : base(session) { }

        string roleName;
        public string RoleName {
            get { return roleName; }
            set { roleName = value; }
        }

        [Association("User-Role", typeof(User))]
        public XPCollection Users {
            get { return GetCollection("Users"); }
        }
    }
}